ERCS/erc-7683.md

```diff
@@ -24,26 +24,47 @@ By implementing a standard, cross-chain intents systems can interoperate and sha
 
 The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
 
-### CrossChainOrder struct
+### Glossary of Terms
 
-A compliant cross-chain order type MUST be ABI decodable into the `CrossChainOrder` type.
+- **Destination Chain**: the chain where the intent is executed and the user receives funds. Note: intents can involve multiple destination chains.
+- **Filler**: a participant who fulfils a user intent on the destination chain(s) and receives payment as a reward.
+- **Leg**: a portion of the user intent that can be executed independently from others. All legs must be executed for an intent to be considered fulfilled.
+- **Origin chain**: the chain where the user sends funds.
+- **Settlement System**: a system that custodies user deposits, verifies fills, and pays fillers for the purpose of faciliating intents.
+- **Settler**: a contract that implements part of the settlement system on a particular chain.
+- **User**: for the purposes of this document, the user is the end-user who is sending the ERC-7683 order.
+
+### Order structs
+
+A compliant cross-chain order type MUST be ABI decodable into either `GaslessCrossChainOrder` or `OnchainCrossChainOrder` type.
 
 ```solidity
-/// @title CrossChainOrder type
-/// @notice Standard order struct to be signed by swappers, disseminated to fillers, and submitted to settlement contracts
-struct CrossChainOrder {
+/// @title GaslessCrossChainOrder CrossChainOrder type
+/// @notice Standard order struct to be signed by users, disseminated to fillers, and submitted to origin settler contracts
+struct GaslessCrossChainOrder {
 	/// @dev The contract address that the order is meant to be settled by.
 	/// Fillers send this order to this contract address on the origin chain
-	address settlementContract;
+	address originSettler;
```

Comment:
Should these addresses also be bytes32 to support non-EVM chains properly? Noticed this in a couple of other places too like the ResolvedCrossChainOrder

Reply:
I don't think so, but let me know if you agree with this logic.

1. 7683 is intended to be _cross compatible_ with other ecosystems via their own standards.
2. Non EVM ecosystems will use their own structs and data formats, so those will not be shared.
3. This field will only be emitted/used on chains that are EVM, since any non-EVM chain would have it's own cross-compatible 7683-like standard.
4. Since this field refers to an address on the same chain as the contract implementing origin settler interface of 7683, we can assume 20 byte addresses are all that's needed.

Summary: we only use bytes32 addresses in cases where the account we're referring to can be on an arbitrary foreign chain, not a chain that we know is hosting contracts that implement 7683. Example of this would be an output token address or output recipient.

Thoughts?

Reply:
Yeah agreed, this makes sense.